File: ApiCredito2-main/GestionIntApi/Models/Admin/MovimientoInventario.cs
using System.ComponentModel.DataAnnotations;

namespace GestionIntApi.Models.Admin
{
    public class MovimientoInventario
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(20)]
        public string TipoMovimiento { get; set; }
        // VALORES: ENTRADA | SALIDA | TRASLADO | VENTA | DEVOLUCION | AJUSTE

        public DateTime FechaMovimiento { get; set; } = DateTime.Now;

        // 🔥 Producto involucrado
        public int ProductoId { get; set; }
        public Producto Producto { get; set; }

        // 🔥 Tienda de ORIGEN (nullable para entradas)
        public int? TiendaOrigenId { get; set; }
        public Tienda? TiendaOrigen { get; set; }

        // 🔥 Tienda de DESTINO (nullable para salidas/ventas)
        public int? TiendaDestinoId { get; set; }
        public Tienda? TiendaDestino { get; set; }

        [MaxLength(500)]
        public string? Observacion { get; set; }

        [MaxLength(100)]
        public string? UsuarioRegistro { get; set; } // Quién hizo el movimiento

        public decimal? MontoVenta { get; set; } // Solo si es VENTA
    }
}

File: CellCompanyFrontend-VersionFinal/src/app/interfaces/MovimientoInventario.ts
import { ProductoBodega } from "./Bodega";
import { Tienda } from "./Tienda";

export interface MovimientoInventario {
  id: number;
  tipoMovimiento: string; // ENTRADA | SALIDA | TRASLADO | VENTA | DEVOLUCION | AJUSTE
  fechaMovimiento: Date;
  productoId: number;
  producto?: ProductoBodega;
  tiendaOrigenId?: number;
  tiendaOrigen?: Tienda;
  tiendaDestinoId?: number;
  tiendaDestino?: Tienda;
  observacion?: string;
  usuarioRegistro?: string;
  montoVenta?: number;
}

export interface TrasladoProducto {
  productoId: number;
  tiendaOrigenId: number;
  tiendaDestinoId: number;
  observacion?: string;
  usuarioRegistro: string;
}
export interface EntradaRequest {
  productoId: number;
  tiendaDestinoId: number;
  observacion?: string;
}

export interface MovimientoHistorial {
  id: number;
  tipoMovimiento: string;
  fechaMovimiento: Date;
  productoId: number;
  productoNombre?: string;
  imei?: string;
  tiendaOrigenId?: number;
  tiendaOrigenNombre?: string;
  tiendaDestinoId?: number;
  tiendaDestinoNombre?: string;
  observacion?: string;
  usuarioRegistro?: string;
  montoVenta?: number;
}

export interface TiendaDestino {
  id: number;
  nombreTienda: string;
  direccion?: string;
  telefono?: string;
  ciudad?: string;
}

 // 👈 Usa tu interfaz existente

export interface StockBodega {
  tiendaId: number;
  tiendaNombre: string;
  totalProductos: number;
  productos: ProductoBodega[]; // 👈 Usa tu interfaz existente
}